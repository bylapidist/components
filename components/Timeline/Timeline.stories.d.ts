import type { StoryObj } from '@storybook/react';
import { BaseProps } from '../types';
import { TimelineProps } from './index';
declare const _default: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@storybook/types").Args>;
export default _default;
export declare const SmallSize: StoryObj<BaseProps & TimelineProps>;
export declare const MediumSize: StoryObj<BaseProps & TimelineProps>;
export declare const LargeSize: StoryObj<BaseProps & TimelineProps>;
