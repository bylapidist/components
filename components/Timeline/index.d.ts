import { BaseProps, SizeType } from '../types';

export interface TimelineItem {
    readonly from: string;
    readonly to: string;
    readonly urlTitle: string;
    readonly url: string;
    readonly title: string;
}
export interface TimelineProps {
    readonly size?: SizeType;
    readonly items: TimelineItem[];
}
declare const Timeline: {
    ({ className, testId, size, items, ...restProps }: BaseProps & TimelineProps): import("react/jsx-runtime").JSX.Element;
    displayName: string;
};
export default Timeline;
