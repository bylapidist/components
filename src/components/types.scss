$levels: 'primary', 'secondary', 'tertiary', 'danger';
$sizes: 'none', 'small', 'medium', 'large';
$families: 'body', 'heading', 'serif', 'mono';
$weights: 'regular', 'medium', 'bold';
