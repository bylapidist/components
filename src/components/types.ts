import {
    ElementType,
    ReactNode,
    CSSProperties,
    ComponentPropsWithoutRef
} from 'react';

export type CommonComponentProps<T extends ElementType> = {
    as?: T;
    testId?: string;
    children?: ReactNode;
    className?: string;
    style?: CSSProperties;
};

export type CommonComponentCombinedProps<
    T extends ElementType,
    K
> = CommonComponentProps<T> &
    Omit<ComponentPropsWithoutRef<T>, keyof CommonComponentProps<T>> &
    K;

export type LevelType = 'primary' | 'secondary' | 'tertiary' | 'danger';

export type SizeType = 'none' | 'small' | 'medium' | 'large';
