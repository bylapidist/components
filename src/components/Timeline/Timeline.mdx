import { Meta, ArgsTable, Story, Source, Canvas } from '@storybook/blocks';
import * as TimelineStories from './Timeline.stories.tsx';

<Meta of={TimelineStories} />

# Timeline

> A primitive for displaying chronological data.

## Example

<Source dark />

## Controlling font size

The `size` prop controls font size. There are three available font sizes, defined as `small`, `medium` and `large`.

### Small

> The `small` size sets the font size to `--font-size-200`

<Canvas>
    <Story of={TimelineStories.SmallSize} />
</Canvas>

### Medium

> The `medium` size sets the font size to `--font-size-300`

<Canvas>
    <Story of={TimelineStories.MediumSize} />
</Canvas>

### Large

> The `large` size sets the font size to `--font-size-400`

<Canvas>
    <Story of={TimelineStories.LargeSize} />
</Canvas>

## Props

<ArgsTable />
