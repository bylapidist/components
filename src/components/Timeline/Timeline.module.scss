@import '../global.module';

.item, .meta {
    display: grid;
    align-items: flex-start;
}

.item {
    grid-template-columns: 1fr 1fr;

    @media screen and (width >= 800px) {
        grid-template-columns: .5fr 1fr;
    }
}

.meta {
    grid-template-columns: 1fr;
}
