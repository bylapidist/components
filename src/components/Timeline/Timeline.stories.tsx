import React from 'react';
import type { Meta, StoryObj, StoryFn } from '@storybook/react';
import { BaseProps } from '../types';
import ThemeProvider from '../ThemeProvider';
import Timeline, { TimelineProps } from './index';

export default {
    title: 'Primitives/Timeline',
    component: Timeline,
    decorators: [(getStory) => <ThemeProvider>{getStory()}</ThemeProvider>]
} as Meta;

const defaultArgs: BaseProps & TimelineProps = {
    items: [
        {
            urlTitle: 'LendInvest',
            title: 'Senior Software Engineer',
            from: 'Apr 2021',
            to: 'Present',
            url: 'https://www.lendinvest.com/'
        },
        {
            urlTitle: 'Golden Charter',
            title: 'Frontend Software Developer',
            from: 'Feb 2020',
            to: 'Apr 2021',
            url: 'https://www.goldencharter.co.uk/'
        }
    ],
    testId: 'Timeline',
    size: 'medium'
};

const DefaultTemplate: StoryFn<BaseProps & TimelineProps> = (args) => (
    <Timeline {...args} />
);

export const SmallSize: StoryObj<BaseProps & TimelineProps> =
    DefaultTemplate.bind({});
SmallSize.args = {
    ...defaultArgs,
    size: 'small'
};
export const MediumSize: StoryObj<BaseProps & TimelineProps> =
    DefaultTemplate.bind({});
MediumSize.args = {
    ...defaultArgs,
    size: 'medium'
};

export const LargeSize: StoryObj<BaseProps & TimelineProps> =
    DefaultTemplate.bind({});
LargeSize.args = {
    ...defaultArgs,
    size: 'large'
};
