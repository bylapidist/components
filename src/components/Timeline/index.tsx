import React from 'react';
import type { BaseProps, SizeType } from '../types';
import Box from '../Box';
import Link from '../Link';
import Text from '../Text';
import styles from './Timeline.module.scss';

export interface TimelineItem {
    readonly from: string;
    readonly to: string;
    readonly urlTitle: string;
    readonly url: string;
    readonly title: string;
}

export interface TimelineProps {
    readonly size?: SizeType;
    readonly items: TimelineItem[];
}

const Timeline = ({
    className,
    testId = 'Timeline',
    size = 'medium',
    items,
    ...restProps
}: BaseProps & TimelineProps) => (
    <Text
        as="ol"
        size={size}
        className={className}
        data-testid={testId}
        {...restProps}
    >
        {items.map((item) => (
            <Box
                as="li"
                key={`${item.urlTitle}-${item.title}`}
                className={styles.item}
                gutter={size}
            >
                <Box gutterY={size}>
                    {item.from} &ndash; {item.to}
                </Box>
                <Box className={styles.meta} gutterY={size}>
                    <Box>{item.title}</Box>
                    <Box>
                        <Link
                            className={styles.urlTitle}
                            href={item.url}
                            size={size}
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            {item.urlTitle}
                        </Link>
                    </Box>
                </Box>
            </Box>
        ))}
    </Text>
);

Timeline.displayName = 'Timeline';
export default Timeline;
