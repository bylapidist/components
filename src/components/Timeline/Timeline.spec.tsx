import React from 'react';
import { describe, expect, test, afterEach } from 'vitest';
import { cleanup, render, screen } from '@testing-library/react';

import { SizeType } from '../types';
import Timeline from './index';

const setup = (Component: React.ReactElement) => render(Component);

const mockItems = [
    {
        urlTitle: 'LendInvest',
        title: 'Senior Software Engineer',
        from: 'Apr 2021',
        to: 'Present',
        url: 'https://www.lendinvest.com/'
    },
    {
        urlTitle: 'Golden Charter',
        title: 'Frontend Software Developer',
        from: 'Feb 2020',
        to: 'Apr 2021',
        url: 'https://www.goldencharter.co.uk/'
    }
];

afterEach(cleanup);

const sizes: SizeType[] = ['small', 'medium', 'large'];
describe('Timeline', () => {
    test('it works with defaults', () => {
        const { container } = setup(<Timeline items={mockItems} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test.each(sizes)('it works with sizes', (size) => {
        const { container } = setup(<Timeline items={mockItems} size={size} />);
        expect(container.firstChild).toMatchSnapshot();
    });

    test('it works with default testId', () => {
        setup(<Timeline items={mockItems} />);
        expect(screen.getByTestId('Timeline')).toBeTruthy();
    });

    test('it works with specified testId', () => {
        setup(<Timeline items={mockItems} testId="TestId" />);
        expect(screen.getByTestId('TestId')).toBeTruthy();
    });

    test('it works with as', () => {
        setup(<Timeline items={mockItems} as="span" />);
        expect(screen.getByTestId('Timeline').nodeName).toBe('SPAN');
    });

    test('it works with className', () => {
        setup(<Timeline items={mockItems} className="test" />);
        expect(
            screen.getByTestId('Timeline').classList.contains('test')
        ).toBeTruthy();
    });
});
