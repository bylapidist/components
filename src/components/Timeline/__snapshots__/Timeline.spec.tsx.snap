// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Timeline > it works with defaults 1`] = `
<ol
  class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6"
  data-testid="Timeline"
>
  <li
    class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26 _item_926827"
    data-testid="Box"
  >
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26"
      data-testid="Box"
    >
      Apr 2021
       – 
      Present
    </div>
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26 _meta_926827"
      data-testid="Box"
    >
      <div
        class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-medium_d6c5f6"
        data-testid="Text"
      >
        Senior Software Engineer
      </div>
      <div
        class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26"
        data-testid="Box"
      >
        <a
          class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6 _this_ef752c _urlTitle_926827"
          data-testid="Link"
          href="https://www.lendinvest.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26 _inner_ef752c"
            data-testid="Link-inner"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-up-right-from-square _icon_ef752c"
              data-icon="arrow-up-right-from-square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                fill="currentColor"
              />
            </svg>
            LendInvest
          </span>
        </a>
      </div>
    </div>
  </li>
  <li
    class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26 _item_926827"
    data-testid="Box"
  >
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26"
      data-testid="Box"
    >
      Feb 2020
       – 
      Apr 2021
    </div>
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26 _meta_926827"
      data-testid="Box"
    >
      <div
        class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-medium_d6c5f6"
        data-testid="Text"
      >
        Frontend Software Developer
      </div>
      <div
        class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26"
        data-testid="Box"
      >
        <a
          class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6 _this_ef752c _urlTitle_926827"
          data-testid="Link"
          href="https://www.goldencharter.co.uk/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26 _inner_ef752c"
            data-testid="Link-inner"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-up-right-from-square _icon_ef752c"
              data-icon="arrow-up-right-from-square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                fill="currentColor"
              />
            </svg>
            Golden Charter
          </span>
        </a>
      </div>
    </div>
  </li>
</ol>
`;

exports[`Timeline > it works with sizes 1`] = `
<ol
  class="_this_d6c5f6 _size-small_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6"
  data-testid="Timeline"
>
  <li
    class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-small_6f2e26 _item_926827"
    data-testid="Box"
  >
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-small_6f2e26"
      data-testid="Box"
    >
      Apr 2021
       – 
      Present
    </div>
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-small_6f2e26 _meta_926827"
      data-testid="Box"
    >
      <div
        class="_this_d6c5f6 _size-small_d6c5f6 _family-body_d6c5f6 _weight-medium_d6c5f6"
        data-testid="Text"
      >
        Senior Software Engineer
      </div>
      <div
        class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26"
        data-testid="Box"
      >
        <a
          class="_this_d6c5f6 _size-small_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6 _this_ef752c _urlTitle_926827"
          data-testid="Link"
          href="https://www.lendinvest.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26 _inner_ef752c"
            data-testid="Link-inner"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-up-right-from-square _icon_ef752c"
              data-icon="arrow-up-right-from-square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                fill="currentColor"
              />
            </svg>
            LendInvest
          </span>
        </a>
      </div>
    </div>
  </li>
  <li
    class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-small_6f2e26 _item_926827"
    data-testid="Box"
  >
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-small_6f2e26"
      data-testid="Box"
    >
      Feb 2020
       – 
      Apr 2021
    </div>
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-small_6f2e26 _meta_926827"
      data-testid="Box"
    >
      <div
        class="_this_d6c5f6 _size-small_d6c5f6 _family-body_d6c5f6 _weight-medium_d6c5f6"
        data-testid="Text"
      >
        Frontend Software Developer
      </div>
      <div
        class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26"
        data-testid="Box"
      >
        <a
          class="_this_d6c5f6 _size-small_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6 _this_ef752c _urlTitle_926827"
          data-testid="Link"
          href="https://www.goldencharter.co.uk/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26 _inner_ef752c"
            data-testid="Link-inner"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-up-right-from-square _icon_ef752c"
              data-icon="arrow-up-right-from-square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                fill="currentColor"
              />
            </svg>
            Golden Charter
          </span>
        </a>
      </div>
    </div>
  </li>
</ol>
`;

exports[`Timeline > it works with sizes 2`] = `
<ol
  class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6"
  data-testid="Timeline"
>
  <li
    class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26 _item_926827"
    data-testid="Box"
  >
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26"
      data-testid="Box"
    >
      Apr 2021
       – 
      Present
    </div>
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26 _meta_926827"
      data-testid="Box"
    >
      <div
        class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-medium_d6c5f6"
        data-testid="Text"
      >
        Senior Software Engineer
      </div>
      <div
        class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26"
        data-testid="Box"
      >
        <a
          class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6 _this_ef752c _urlTitle_926827"
          data-testid="Link"
          href="https://www.lendinvest.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26 _inner_ef752c"
            data-testid="Link-inner"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-up-right-from-square _icon_ef752c"
              data-icon="arrow-up-right-from-square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                fill="currentColor"
              />
            </svg>
            LendInvest
          </span>
        </a>
      </div>
    </div>
  </li>
  <li
    class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26 _item_926827"
    data-testid="Box"
  >
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26"
      data-testid="Box"
    >
      Feb 2020
       – 
      Apr 2021
    </div>
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-medium_6f2e26 _meta_926827"
      data-testid="Box"
    >
      <div
        class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-medium_d6c5f6"
        data-testid="Text"
      >
        Frontend Software Developer
      </div>
      <div
        class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26"
        data-testid="Box"
      >
        <a
          class="_this_d6c5f6 _size-medium_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6 _this_ef752c _urlTitle_926827"
          data-testid="Link"
          href="https://www.goldencharter.co.uk/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26 _inner_ef752c"
            data-testid="Link-inner"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-up-right-from-square _icon_ef752c"
              data-icon="arrow-up-right-from-square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                fill="currentColor"
              />
            </svg>
            Golden Charter
          </span>
        </a>
      </div>
    </div>
  </li>
</ol>
`;

exports[`Timeline > it works with sizes 3`] = `
<ol
  class="_this_d6c5f6 _size-large_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6"
  data-testid="Timeline"
>
  <li
    class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-large_6f2e26 _item_926827"
    data-testid="Box"
  >
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-large_6f2e26"
      data-testid="Box"
    >
      Apr 2021
       – 
      Present
    </div>
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-large_6f2e26 _meta_926827"
      data-testid="Box"
    >
      <div
        class="_this_d6c5f6 _size-large_d6c5f6 _family-body_d6c5f6 _weight-medium_d6c5f6"
        data-testid="Text"
      >
        Senior Software Engineer
      </div>
      <div
        class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26"
        data-testid="Box"
      >
        <a
          class="_this_d6c5f6 _size-large_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6 _this_ef752c _urlTitle_926827"
          data-testid="Link"
          href="https://www.lendinvest.com/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26 _inner_ef752c"
            data-testid="Link-inner"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-up-right-from-square _icon_ef752c"
              data-icon="arrow-up-right-from-square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                fill="currentColor"
              />
            </svg>
            LendInvest
          </span>
        </a>
      </div>
    </div>
  </li>
  <li
    class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-large_6f2e26 _item_926827"
    data-testid="Box"
  >
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-large_6f2e26"
      data-testid="Box"
    >
      Feb 2020
       – 
      Apr 2021
    </div>
    <div
      class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-large_6f2e26 _meta_926827"
      data-testid="Box"
    >
      <div
        class="_this_d6c5f6 _size-large_d6c5f6 _family-body_d6c5f6 _weight-medium_d6c5f6"
        data-testid="Text"
      >
        Frontend Software Developer
      </div>
      <div
        class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26"
        data-testid="Box"
      >
        <a
          class="_this_d6c5f6 _size-large_d6c5f6 _family-body_d6c5f6 _weight-regular_d6c5f6 _this_ef752c _urlTitle_926827"
          data-testid="Link"
          href="https://www.goldencharter.co.uk/"
          rel="noopener noreferrer"
          target="_blank"
        >
          <span
            class="_this_6f2e26 _elevation-none_6f2e26 _roundness-none_6f2e26 _gutter-none_6f2e26 _gutterX-none_6f2e26 _gutterY-none_6f2e26 _inner_ef752c"
            data-testid="Link-inner"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-arrow-up-right-from-square _icon_ef752c"
              data-icon="arrow-up-right-from-square"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 0c-17.7 0-32 14.3-32 32s14.3 32 32 32l82.7 0L201.4 265.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L448 109.3l0 82.7c0 17.7 14.3 32 32 32s32-14.3 32-32l0-160c0-17.7-14.3-32-32-32L320 0zM80 32C35.8 32 0 67.8 0 112L0 432c0 44.2 35.8 80 80 80l320 0c44.2 0 80-35.8 80-80l0-112c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 112c0 8.8-7.2 16-16 16L80 448c-8.8 0-16-7.2-16-16l0-320c0-8.8 7.2-16 16-16l112 0c17.7 0 32-14.3 32-32s-14.3-32-32-32L80 32z"
                fill="currentColor"
              />
            </svg>
            Golden Charter
          </span>
        </a>
      </div>
    </div>
  </li>
</ol>
`;
